src/pkg/packager/deploy.go

```diff
@@ -359,6 +356,10 @@ func (p *Packager) deployComponent(ctx context.Context, component types.ZarfComp
 		return charts, fmt.Errorf("unable to run component after action: %w", err)
 	}
 
+	err = g.Wait()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Is there a reason we are calling `g.Wait()` here rather than immediately after looping over DataInjections and calling `HandleDataInjection`?

Reply:
Mostly to reflect the logic which was present previously. The impact of doing it this way is that the actions may run during data injection is under way. Maybe we don't want this behavior at all?